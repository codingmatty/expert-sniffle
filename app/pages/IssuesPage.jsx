import React from 'react';

import ReposList from 'components/ReposList';
import IssuesList from 'components/IssuesList';

import './IssuesPage.scss';

export default function IssuesPage({ repoId }) {
  return (
    <div className="issues-page flex direction-row mx-4 medium:mx-auto">
      <ReposList selectedRepoId={repoId} />
      <IssuesList selectedRepoId={repoId} />
    </div>
  );
}
