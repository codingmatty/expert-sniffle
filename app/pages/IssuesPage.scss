.issues-page {
  .repos-list {
    display: none;
  }

  @media (min-width: 768px) {
    max-width: 738px;
    .back-to-repos {
      display: none;
    }
    .repos-list {
      display: block;
      max-width: 300px;
    }
    .issues-list {
      margin-left: 1rem;
    }
  }

  @media (min-width: 1030px) {
    max-width: 1000px;
  }
}
