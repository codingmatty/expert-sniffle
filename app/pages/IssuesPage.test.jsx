import '@testing-library/jest-dom';
import configureStore from 'redux-mock-store';
import React from 'react';
import { render, screen } from '@testing-library/react';
import IssuesPage from './IssuesPage';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';

const mockStore = configureStore([thunk]);

describe('IssuesPage', () => {
  describe('loading', () => {
    test('shows loading message', () => {
      const { asFragment } = render(
        <Provider
          store={mockStore({ repos: { data: [] }, issues: { loading: true } })}
        >
          <IssuesPage repoId="123456789" />
        </Provider>
      );
      expect(screen.getByText('Loading...')).toBeDefined();
      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('with data', () => {
    test('empty array', () => {
      const { asFragment } = render(
        <Provider
          store={mockStore({ repos: { data: [] }, issues: { data: [] } })}
        >
          <IssuesPage repoId="123456789" />
        </Provider>
      );
      expect(
        screen.getByText(
          'No issues found for 123456789.Try selecting a different repo.'
        )
      ).toBeDefined();
      expect(asFragment()).toMatchSnapshot();
    });

    test('with list of repos', () => {
      const repos = [
        {
          id: 123456789,
          name: 'Name',
          full_name: 'full/name',
          description: 'Lorem Ipsum Dolor Sit Amet',
          stargazers_count: 1,
          open_issues_count: 3,
          html_url: 'https://github.com/full/name',
        },
      ];
      const data = [
        {
          id: 1,
          newIssue: true,
          title: 'Issue #1',
          user: {
            avatar_url: 'test.com',
          },
        },
        {
          id: 2,
          newIssue: false,
          title: 'Issue #2',
          user: {
            avatar_url: 'test.com',
          },
        },
        {
          id: 3,
          newIssue: false,
          title: 'Issue #3',
          user: {
            avatar_url: 'test.com',
          },
        },
      ];
      const { asFragment } = render(
        <Provider
          store={mockStore({ repos: { data: repos }, issues: { data } })}
        >
          <IssuesPage repoId="123456789" />
        </Provider>
      );
      expect(screen.getByTestId('repo-details')).toHaveClass('selected');
      expect(screen.getAllByTestId('issue-details')).toHaveLength(3);
      expect(screen.getAllByTestId('issue-details')[0]).toHaveClass('new');
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
