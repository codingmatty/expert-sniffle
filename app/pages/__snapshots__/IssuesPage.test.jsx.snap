// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IssuesPage loading shows loading message 1`] = `
<DocumentFragment>
  <div
    class="issues-page flex direction-row mx-4 medium:mx-auto"
  >
    <div
      class="repos-list grow-1"
    >
      <div
        class="flex direction-row space-between items-center mb-2"
      >
        <div
          class="flex direction-column"
        >
          <h2
            class="mb-2"
          >
            Repos
          </h2>
          <a
            aria-current="page"
            href="/"
          >
            Update Token
          </a>
        </div>
        <button>
          Reload Repos
        </button>
      </div>
      <ul>
        <p>
          No Repos Found for provided token
        </p>
      </ul>
    </div>
    <div
      class="issues-list grow-1"
    >
      <a
        class="back-to-repos mt-4 flex items-center"
        href="/repos"
      >
        <i
          class="material-icons mr-2 decoration-none"
        >
          arrow_back
        </i>
        Back To Repos
      </a>
      <div
        class="flex direction-row space-between items-center"
      >
        <h2>
          Issues
        </h2>
        <button>
          Reload Issues
        </button>
      </div>
      <p>
        Loading...
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`IssuesPage with data empty array 1`] = `
<DocumentFragment>
  <div
    class="issues-page flex direction-row mx-4 medium:mx-auto"
  >
    <div
      class="repos-list grow-1"
    >
      <div
        class="flex direction-row space-between items-center mb-2"
      >
        <div
          class="flex direction-column"
        >
          <h2
            class="mb-2"
          >
            Repos
          </h2>
          <a
            aria-current="page"
            href="/"
          >
            Update Token
          </a>
        </div>
        <button>
          Reload Repos
        </button>
      </div>
      <ul>
        <p>
          No Repos Found for provided token
        </p>
      </ul>
    </div>
    <div
      class="issues-list grow-1"
    >
      <a
        class="back-to-repos mt-4 flex items-center"
        href="/repos"
      >
        <i
          class="material-icons mr-2 decoration-none"
        >
          arrow_back
        </i>
        Back To Repos
      </a>
      <div
        class="flex direction-row space-between items-center"
      >
        <h2>
          Issues
        </h2>
        <button>
          Reload Issues
        </button>
      </div>
      <p
        data-testid="empty-issues-message"
      >
        No issues found for 123456789.
        <br />
        <br />
        Try selecting a different repo.
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`IssuesPage with data with list of repos 1`] = `
<DocumentFragment>
  <div
    class="issues-page flex direction-row mx-4 medium:mx-auto"
  >
    <div
      class="repos-list grow-1"
    >
      <div
        class="flex direction-row space-between items-center mb-2"
      >
        <div
          class="flex direction-column"
        >
          <h2
            class="mb-2"
          >
            Repos
          </h2>
          <a
            aria-current="page"
            href="/"
          >
            Update Token
          </a>
        </div>
        <button>
          Reload Repos
        </button>
      </div>
      <ul>
        <li>
          <div
            class="repo-details pointer space-between mt-4 border border-gray radius-3 p-4 bg-white selected"
            data-testid="repo-details"
          >
            <div
              class="info"
            >
              <div
                class="text-large"
              >
                <b>
                  Name
                </b>
              </div>
              <div>
                <a
                  href="https://github.com/full/name"
                  target="_blank"
                >
                  full/name 
                  <i
                    class="material-icons "
                  >
                    open_in_new
                  </i>
                </a>
              </div>
            </div>
            <div
              class="star-count flex items-center text-large"
            >
              <i
                class="material-icons mr-2"
              >
                star
              </i>
              1
            </div>
            <div
              class="issue-count flex items-center text-large"
            >
              <i
                class="material-icons mr-2"
              >
                error_outline
              </i>
              3
            </div>
            <p
              class="mt-4 description"
            >
              Lorem Ipsum Dolor Sit Amet
            </p>
          </div>
        </li>
      </ul>
    </div>
    <div
      class="issues-list grow-1"
    >
      <a
        class="back-to-repos mt-4 flex items-center"
        href="/repos"
      >
        <i
          class="material-icons mr-2 decoration-none"
        >
          arrow_back
        </i>
        Back To Repos
      </a>
      <div
        class="flex direction-row space-between items-center"
      >
        <h2>
          Issues
        </h2>
        <button>
          Reload Issues
        </button>
      </div>
      <h3>
        full/name
      </h3>
      <ul>
        <li
          class="grabbable"
          data-id="1"
        >
          <div
            class="issue-details mt-4 border border-gray radius-3 p-4 bg-white new"
            data-testid="issue-details"
          >
            <div
              class="flex space-between"
            >
              <img
                class="radius-3"
                height="40"
                src="test.com&s=40"
                width="40"
              />
              <div
                class="text-small"
              >
                <div>
                  Created: 12/31/2019
                </div>
                <div>
                  Last updated a few seconds ago
                </div>
              </div>
            </div>
            <p
              class="mt-4"
            >
              Issue #1
            </p>
          </div>
        </li>
        <li
          class="grabbable"
          data-id="2"
        >
          <div
            class="issue-details mt-4 border border-gray radius-3 p-4 bg-white "
            data-testid="issue-details"
          >
            <div
              class="flex space-between"
            >
              <img
                class="radius-3"
                height="40"
                src="test.com&s=40"
                width="40"
              />
              <div
                class="text-small"
              >
                <div>
                  Created: 12/31/2019
                </div>
                <div>
                  Last updated a few seconds ago
                </div>
              </div>
            </div>
            <p
              class="mt-4"
            >
              Issue #2
            </p>
          </div>
        </li>
        <li
          class="grabbable"
          data-id="3"
        >
          <div
            class="issue-details mt-4 border border-gray radius-3 p-4 bg-white "
            data-testid="issue-details"
          >
            <div
              class="flex space-between"
            >
              <img
                class="radius-3"
                height="40"
                src="test.com&s=40"
                width="40"
              />
              <div
                class="text-small"
              >
                <div>
                  Created: 12/31/2019
                </div>
                <div>
                  Last updated a few seconds ago
                </div>
              </div>
            </div>
            <p
              class="mt-4"
            >
              Issue #3
            </p>
          </div>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
